crates/sui-storage/src/event_store/sql.rs

```diff
@@ -398,6 +413,7 @@ impl EventStore for SqlEventStore {
                 .execute(&self.pool)
                 .await
                 .map_err(convert_sqlx_err)?;
+            event.seq_num = res.last_insert_rowid() as u64;
```

Comment:
Ok, so modifying the input events is a really bad idea.  It makes the API non idempotent and its side effects could introduce lots of bugs.  

A second problem is that getting the last inserted rowID for every single row might be a problem when we move to bulk inserts, which I'm likely to do soon for performance reasons.

Instead, can we simply return the last row ID that was generated from the database from the method?    
Also, I'm not sure if we should be feeding back the sequence number from the event store into the other events.  What is the point of that?  The sequence number is kind of an internal thing to implement paging, I'm questioning if it should be an external API.

Reply:
we really only need a unique ID here, the unique id is needed for the pagination cursor so will need to exposes it to the end user, maybe we can use digest or a random hash instead of an incremental index number, WDYT?

Reply:
I have removed the ID from EventEnvelop, it's now only used for pagination cursor